apps/src/javalab/JavalabEditor.jsx

```diff
@@ -622,6 +642,15 @@ class JavalabEditor extends React.Component {
           saveButtonText="Create"
           errorMessage={newFileError}
         />
+        <CommitDialog
```

Comment:
[Non-blocking comment] It seems like a lot of JavalabEditor is now logic & display code around dialogs. Would it make sense to break out a new JavalabDialog component to contain all this logic & display code?

Reply:
yes, i think so! i won't have time to do that right now with the video deadline -- what's the best way to log/prioritize this work?